src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -133,10 +120,16 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        agent = self.get(self._USER_AGENT, '')
-        user_agent = user_agent.strip()
-        if user_agent not in agent:
-            self[self._USER_AGENT] = f"{agent} {user_agent}".strip()
+        if "user_agent" in self:
+            if user_agent not in self.user_agent:
+                self.user_agent = f"{self.user_agent} {user_agent}"
+        else:
+            self.user_agent = user_agent
```

Comment:
```python
    if "user_agent" in self:  
        if user_agent not in self.user_agent:  
            self.user_agent = f"{self.user_agent.strip()} {user_agent.strip()}"  
    else:  
        self.user_agent = user_agent.strip()  
```
May be a better way to strip() when append. Pending your choice.

Reply:
due to context['user_agent'] = ' xxxx ' will will bypass this function.
So get_user_agent function muse be add strip.

Reply:
Yes. I understand. It seems that `strip()` in `get_user_agent` is a must-have action.

Reply:
I also added a strip here.